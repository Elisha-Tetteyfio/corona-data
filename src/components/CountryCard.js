import React from 'react';
import styles from './CountryCard.module.css';
import icon from '../images/details.png';

const CountryCard = (country) => {
  const { name, latest_data: latestData = { confirmed: 0 } } = country;
  return (
    <div className={styles.container}>
      <div><img src={icon} alt="details" /></div>
      <div className={styles.bottom}>
        <div className={styles.name}>{name}</div>
        <div>{latestData.confirmed}</div>
      </div>
    </div>
  );
};

export default CountryCard;
