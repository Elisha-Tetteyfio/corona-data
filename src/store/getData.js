import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const url = 'https://corona-api.com/countries';
const GET_DATA = 'coronoa-data/GET_DATA';

export const fetchData = createAsyncThunk(GET_DATA, async () => {
  const result = await fetch(url);
  const resultJson = await result.json();
  return resultJson.data;
});

const dataSlice = createSlice({
  name: 'corona-data',
  initialState: [{ name: '', code: '', latest_data: { confirmed: '', deaths: '', recovered: '' } }],
  extraReducers: {
    [fetchData.fulfilled]: (state, action) => action.payload,
  },
});

// export const { } = dataSlice.actions;
export default dataSlice.reducer;
