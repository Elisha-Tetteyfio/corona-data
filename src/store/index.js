import { configureStore } from '@reduxjs/toolkit';
import data from './getData';

const store = configureStore({
  reducer: {
    data,
  },
});

export default store;
