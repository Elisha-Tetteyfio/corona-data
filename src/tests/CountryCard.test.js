import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../store/index';
import CountryCard from '../components/CountryCard';

describe('Testing the components', () => {
  it('test CountryCard', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CountryCard />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
