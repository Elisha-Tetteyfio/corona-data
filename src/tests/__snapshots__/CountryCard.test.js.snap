// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the components test CountryCard 1`] = `
<div
  className="container"
>
  <div>
    <img
      alt="details"
      src="details.png"
    />
  </div>
  <div
    className="bottom"
  >
    <div
      className="name"
    />
    <div>
      0
    </div>
  </div>
</div>
`;
